tools/clang/unittests/SPIRV/SpirvConstantTest.cpp

```diff
@@ -0,0 +1,36 @@
+//===- unittests/SPIRV/SpirvConstantTest.cpp --- SPIR-V Constant tests ----===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "clang/SPIRV/SpirvInstruction.h"
+
+namespace {
+using namespace clang::spirv;
+
+TEST(SpirvConstant, Float16) {
+  const uint16_t f16 = 12;
+  SpirvConstantFloat constant(f16, {}, 0, {});
+  EXPECT_EQ(f16, constant.getValue16());
+}
+
+TEST(SpirvConstant, Float32) {
+  const float f32 = 1.5;
+  SpirvConstantFloat constant(f32, {}, 0, {});
+  EXPECT_EQ(f32, constant.getValue32());
+}
+
+TEST(SpirvConstant, Float64) {
+  const double f64 = 3.14;
+  SpirvConstantFloat constant(f64, {}, 0, {});
+  EXPECT_EQ(f64, constant.getValue64());
+}
```

Comment:
Would be nice to add cases for integers (bitwidth, signedness) too.

Reply:
That would require creating signed or unsigned integer QualTypes. I'm not sure how to do that... That'd require the ASTContext?

Reply:
Good point. Sorry missed that. Yeah. This is also another reason that we probably don't want to rely on AST type in SPIR-V type. But it can be done later.